runelite-client/src/main/java/net/runelite/client/plugins/driftnet/DriftNetOverlay.java

```diff
@@ -0,0 +1,109 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.driftnet;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Shape;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.Point;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+class DriftNetOverlay extends Overlay
+{
+	private final DriftNetConfig config;
+	private final DriftNetPlugin plugin;
+
+	@Inject
+	private DriftNetOverlay(DriftNetConfig config, DriftNetPlugin plugin)
+	{
+		this.config = config;
+		this.plugin = plugin;
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.LOW);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!plugin.isInDriftNetArea())
+		{
+			return null;
+		}
+
+		if (config.highlightUntaggedFish())
+		{
+			renderFish(graphics);
+		}
+		if (config.showNetStatus())
+		{
+			renderNets(graphics);
+		}
+		return null;
+	}
+
+	private void renderFish(Graphics2D graphics)
+	{
+		for (NPC fish : plugin.getFish())
+		{
+			if (!plugin.getTaggedFish().containsKey(fish))
+			{
+				OverlayUtil.renderActorOverlay(graphics, fish, "", config.untaggedFishColor());
+			}
+		}
+	}
+
+	private void renderNets(Graphics2D graphics)
+	{
+		for (DriftNet net : plugin.getNETS())
+		{
+			if (net.getNet() == null && net.getStatus() != null)
```

Comment:
what is this trying to do?

Reply:
It was going to prevent NPEs during the overlay rendering, but i don't know how i got it that wrong. Sorry.
It should be `net.getNet() == null || net.getStatus() == null`. Maybe it's not needed if it's been like this during testing,  i'll look into it later today.